### DISCUSSION TITLE
Tool rejected.

### DISCUSSION THREAD

**Discussion by satishreddyv** [MscrmTools/XrmToolBox#1095] [General]
Hello... First of all, a big thank you, and I cannot say how much this tool is helpful for our team and Dynamics 365 community.

I have submitted a tool called "Silwood Dynamics 365 Metadata Exporter" which has been rejected twice. I think I have followed all the guidelines you mentioned in the documentation. Is there a way to get feedback on why it was rejected?

Once again thank you for your time :)

**Comment by MscrmTools** [comment]
Hi,
You should have received an email with the reason it was rejected. Please check your spam folder.
In the meantime, here are the reasons:

- Assembly version mismatch with Nuget package. Both must have the same version
- Missing dependency : Microsoft.Azure.Cosmos.Table, Version=1.0.8.0

**Comment by satishreddyv** [comment]
Thank you, I will fix the issue and submit it. However, I couldn't find the email anywhere including the spam folder. I double-checked but couldn't find it.

**Comment by satishreddyv** [comment]
Not sure why missing dependency error. It was already added. However, it was also part of the Nuget Download cache. I just removed the package from the cache.
![image](https://user-images.githubusercontent.com/37939824/147430700-dabed72a-380e-436d-b352-e868aff8e3be.png)

**Comment by satishreddyv** [comment]
@MscrmTools , 

Hope you are doing well. Could you please review my updated tool? I have updated it a week ago. Apologies for wasting your time again. 

If you are going to reject the tool again, is it possible for me to hire you for a couple of hours? I may need help in packaging it. I am not sure what I am missing. I followed all the steps mentioned in the documentation.

Appreciate your time and help.

Thanks
Satish

**Comment by MscrmTools** [comment]
Hi,
So you did merge the missing assembly, that is good. But now, the assembly your merged also has dependencies. You have to merge all assemblies required to make your tool works.
So far, XrmToolBox is showing that the following assemblies are missing:

- Microsoft.Azure.DocumentDB.Core
- Microsoft.OData.Core

Try to merge these one too

And you don't need me to validate it works. Simply put your merged assembly in an XrmToolBox installation and try to run XrmToolBox, it will tell you if a dependency is missing

**Comment by satishreddyv** [comment]
Thank you very much for your reply, I just submitted an updated one without any external dependencies. I tested myself by loading it. Thank you.
